src/components/textbox/textbox.test.js

```diff
@@ -0,0 +1,712 @@
+          .and("have.css", "font-weight", "900")
+          .and("have.css", "margin-right", "8px");
+      }
+    );
+
+    it.each([
+      [true, "input", "label"],
+      [false, "label", "input"],
+    ])(
+      "should render Textbox with reverse prop set to %s",
+      (boolean, firstElement, secondElement) => {
+        CypressMountWithProviders(<TextboxComponent reverse={boolean} />);
+
+        cyRoot()
+          .children()
+          .children()
+          .children()
+          .children(0)
+          .find(firstElement);
+        cyRoot()
+          .children()
+          .children()
+          .children()
+          .children(1)
+          .find(secondElement);
+      }
+    );
+
+    it.each([
+      ["small", "32px"],
+      ["medium", "40px"],
+      ["large", "48px"],
+    ])(
+      "should use %s as size and render Textbox with %s as height",
+      (size, height) => {
+        CypressMountWithProviders(<TextboxComponent size={size} />);
+
+        textbox().should("have.css", "min-height", height);
+      }
+    );
+
+    it("should render Textbox with positionedChildren prop", () => {
+      CypressMountWithProviders(
+        <TextboxComponent positionedChildren={<Button>Test</Button>} />
+      );
+
+      textbox()
+        .parent()
+        .children(0)
+        .should("have.attr", "data-component", "button")
+        .and("be.visible");
+    });
+
+    it.each([
+      ["flex", "399"],
+      ["flex", "400"],
+      ["block", "401"],
+    ])(
+      "should check Textbox label alignment is %s with adaptiveLabelBreakpoint %s and viewport 400",
+      (displayValue, breakpoint) => {
+        cy.viewport(400, 300);
+
+        CypressMountWithProviders(
+          <TextboxComponent adaptiveLabelBreakpoint={breakpoint} />
+        );
+
+        getDataElementByValue("label")
+          .parent()
+          .parent()
+          .should("have.css", "display", displayValue);
+      }
+    );
+
+    it("should render Textbox with labelId prop", () => {
+      CypressMountWithProviders(
+        <TextboxComponent
+          id={`${CHARACTERS.STANDARD}_cy`}
+          labelId={CHARACTERS.STANDARD}
+        />
+      );
+
+      getDataElementByValue("label")
+        .should("have.attr", "id", CHARACTERS.STANDARD)
+        .and("have.attr", "for", `${CHARACTERS.STANDARD}_cy`);
+    });
+
+    it.each([CHARACTERS.SPECIALCHARACTERS, CHARACTERS.DIACRITICS])(
+      "should render Textbox with fieldHelp prop set to %s",
+      (fieldHelp) => {
+        CypressMountWithProviders(<TextboxComponent fieldHelp={fieldHelp} />);
+
+        fieldHelpPreview().should("have.text", fieldHelp);
+      }
+    );
+
+    it("should render Textbox with formattedValue prop", () => {
+      CypressMountWithProviders(
+        <TextboxComponent formattedValue={CHARACTERS.STANDARD} />
+      );
+
+      textboxInput().should("have.attr", "value", CHARACTERS.STANDARD);
+    });
+
+    it.each(["add", "filter", "play"])(
```

Comment:
Can we list a few more elements here like "arrow_left_boxed" , "bullet_list_dotted", etc...?

Reply:
what it is the point of that ?